### ISSUE TITLE
Color Scheme Error: `configparser.NoOptionError: No option 'custom-0/normal' in section: 'appearance'`

### ISSUE THREAD

**Issue by RayrayRaven** [spyder-ide/spyder#22492]
### What steps will reproduce the problem?

made a custom color scheme and accidenntly deleted the scheme while still using it.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\plugins\preferences\plugin.py", line 280, in <lambda>
    lambda: self.open_dialog(main.prefs_dialog_size))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\plugins\preferences\plugin.py", line 260, in open_dialog
    container.create_dialog(
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\plugins\preferences\widgets\container.py", line 58, in create_dialog
    page.initialize()
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\plugins\preferences\api.py", line 86, in initialize
    self.setup_page()
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\plugins\appearance\confpage.py", line 164, in setup_page
    self.scheme_editor_dialog.add_color_scheme_stack(name, custom=True)
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\plugins\appearance\widgets.py", line 128, in add_color_scheme_stack
    value = self.parent.get_option(option)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\api\plugins\new_api.py", line 498, in get_conf
    return self._conf.get(section, option, default)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\config\manager.py", line 499, in get
    value = config.get(section=section, option=option, default=default)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\config\user.py", line 943, in get
    return config.get(section=section, option=option, default=default)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\racha\anaconda3\Lib\site-packages\spyder\config\user.py", line 498, in get
    raise cp.NoOptionError(option, section)
configparser.NoOptionError: No option 'custom-0/normal' in section: 'appearance'
```

## Versions

* Spyder version: 5.5.1  (conda)
* Python version: 3.12.4 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.10
* Operating System: Windows-11-10.0.22631-SP0

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0             :  1.4.0 (OK)
chardet >=2.0.0                  :  4.0.0 (OK)
cloudpickle >=0.5.0              :  2.2.1 (OK)
cookiecutter >=1.6.0             :  2.6.0 (OK)
diff_match_patch >=20181111      :  20200713 (OK)
intervaltree >=3.0.2             :  3.1.0 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1 :  8.25.0 (OK)
jedi >=0.17.2,<0.20.0            :  0.18.1 (OK)
jellyfish >=0.7                  :  1.0.1 (OK)
jsonschema >=3.2.0               :  4.19.2 (OK)
keyring >=17.0.0                 :  24.3.1 (OK)
nbconvert >=4.0                  :  7.10.0 (OK)
numpydoc >=0.6.0                 :  1.7.0 (OK)
paramiko >=2.4.0                 :  2.8.1 (OK)
parso >=0.7.0,<0.9.0             :  0.8.3 (OK)
pexpect >=4.4.0                  :  4.8.0 (OK)
pickleshare >=0.4                :  0.7.5 (OK)
psutil >=5.3                     :  5.9.0 (OK)
pygments >=2.0                   :  2.15.1 (OK)
pylint >=2.5.0,<3.1              :  2.16.2 (OK)
pylint_venv >=3.0.2              :  3.0.3 (OK)
pyls_spyder >=0.4.0              :  0.4.0 (OK)
pylsp >=1.10.0,<1.11.0           :  1.10.0 (OK)
pylsp_black >=2.0.0,<3.0.0       :  2.0.0 (OK)
qdarkstyle >=3.2.0,<3.3.0        :  3.2.3 (OK)
qstylizer >=0.2.2                :  0.2.2 (OK)
qtawesome >=1.2.1                :  1.2.2 (OK)
qtconsole >=5.5.1,<5.6.0         :  5.5.1 (OK)
qtpy >=2.1.0                     :  2.4.1 (OK)
rtree >=0.9.7                    :  1.0.1 (OK)
setuptools >=49.6.0              :  69.5.1 (OK)
sphinx >=0.6.6                   :  7.3.7 (OK)
spyder_kernels >=2.5.0,<2.6.0    :  2.5.0 (OK)
textdistance >=4.2.0             :  4.2.1 (OK)
three_merge >=0.1.1              :  0.1.1 (OK)
watchdog >=0.10.3                :  4.0.1 (OK)
zmq >=22.1.0                     :  25.1.2 (OK)

# Optional:
cython >=0.21                    :  None (NOK)
matplotlib >=3.0.0               :  3.8.4 (OK)
numpy >=1.7                      :  1.26.4 (OK)
pandas >=1.1.1                   :  2.2.2 (OK)
scipy >=0.17.0                   :  1.13.1 (OK)
sympy >=0.7.3                    :  1.12 (OK)
```

**Comment by dalthviz** [comment]
Hi @RayrayRaven thank you for the report and sorry for the late response! Seems like this could be a duplicate of #20531. Could it be possible for you to share over #20531 the steps you followed to experience the issue/see the error? It could be quite useful if you could create a GIF or video showing the steps for clarity. Any further info in order to reproduce this is greatly appreciated. Let us know!

**Comment by ccordoba12** [comment]
@dalthviz, according to what @RayrayRaven said, I think what's happening is this:

* Users create a custom color scheme, apply it and close Preferences.
* They open Preferences again and delete that scheme without selecting a new one, and close it.
* When opening Preferences one more time, the error shows up.

Could you try those steps and see if you're able to reproduce the problem? Thanks!

**Comment by dalthviz** [comment]
I was unable to reproduce the problem following those steps using Spyder 6.0.1 from the Windows installer:

![color_schema_steps](https://github.com/user-attachments/assets/c32673b5-c8b8-46f2-8790-2f2cc478e51d)

When deleting the custom color schema the selected value fallbacks so no issue is raised. However, I noticed that changing the color schema only affects new opened files when using a custom one? Is that expected or should I create another issue for that? Also, seems like when it fallbacks to a schema available the Editor updates even without calling apply (which seems like then some logic is there to prevent issues like this one 🤔)

**Comment by ccordoba12** [comment]
Ok, thanks @dalthviz for the confirmation. What if you do this?

* Create a custom color scheme and apply it, but don't close Preferences
* Remove it and close Preferences
* Open Preferences again

You also said:

> However, I noticed that changing the color schema only affects new opened files when using a custom one? Is that expected or should I create another issue for that?

That's wrong, so yep, please open a new issue about it.

**Comment by dalthviz** [comment]
>  What if you do this?
> 
> * Create a custom color scheme and apply it, but don't close Preferences
> * Remove it and close Preferences
> * Open Preferences again

Checked those steps but still no error is raised:

![custom_syntax](https://github.com/user-attachments/assets/8407b39b-7b6a-4e05-8209-cae8e4dd2dd2)

**Comment by dalthviz** [comment]
Well I guess handling things with a try-catch could work as a workaround :+1:. However, checking the traceback and related code, seems like the error triggers when the config page is being build and the custom color schemes are being loaded so some how is like although a color scheme is listed over the appearance `custom_names` option, its values doesn't exist anymore 🤔 If handling this via a try-catch statement is enough, I would say that such handling should be done over the confpage logic that is loading the color schemes so around:

https://github.com/spyder-ide/spyder/blob/3274e793422497da39940e8ac4dad3ef793acafe/spyder/plugins/appearance/confpage.py#L216-L217

And over the catch part we should not only ensure that the selected scheme is not the faulty one but also that its entry is not part of the options available over the scheme combobox (probably by removing it from the `custom_names`  option before calling `update_combobox` I think)? 🤔

**Comment by ccordoba12** [comment]
> If handling this via a try-catch statement is enough, I would say that such handling should be done over the confpage logic that is loading the color schemes

Ok, things doesn't seem so simple as I thought.

These are a couple of resources that could help to solve this problem:

* https://stackoverflow.com/questions/14129278/how-do-i-customize-text-color-in-ipython
* https://stackoverflow.com/questions/47953896/syntax-colors-on-ipython

**Comment by dalthviz** [comment]
Should we close the related issues about this like #20531 and #5790 @ccordoba12 in favor of this one? Also, should we mark this with the `help wanted` label?

Besides that, for tracebility, other already closed but related issues are #13006, #10390, #7064

**Comment by ccordoba12** [comment]
> Should we close the related issues about this like https://github.com/spyder-ide/spyder/issues/20531 and https://github.com/spyder-ide/spyder/issues/5790 @ccordoba12 in favor of this one?

Yep, let's do that.

> Also, should we mark this with the help wanted label?

No, I think we should solve it ourselves now that we have a good idea of what it needs to be done.
